### PR TITLE
Add StatusCategory resource

### PR DESCRIPTION
**Pull request by slank**
Add StatusCategory as a resource

### CODE REVIEW

**Comment by slank** [comment]
The failing check doesn't appear related to my changes. If I'm mistaken, just let me know what I missed.

File: jira/client.py [hunk 1]
```diff
@@ -2142,6 +2143,22 @@ def status(self, id):
         """
         return self._find_for_resource(Status, id)
 
+    # Category
+
+    def statuscategories(self):
+        """Get a list of status category Resources from the server."""
+        r_json = self._get_json('statuscategory')
+        statuscategories = [StatusCategory(self._options, self._session, raw_stat_json)
+                    for raw_stat_json in r_json]
```
Comment (hdost) [inline jira/client.py]: This is what travis CI is complaining about in the latest build
```
./jira/client.py:2152:21: E128 continuation line under-indented for visual indent
1       E128 continuation line under-indented for visual indent
```

**Comment by mwedgwood-rmn** [comment]
I've fixed the formatting issues, amended, and pushed.

**Comment by mwedgwood-rmn** [comment]
@ssbarnea can you explain what needs to change? I believe I've already addressed the changes requested by @hdost

**Comment by hdost** [comment]
@mwedgwood-rmn The only think I can think of is rebasing off of master. (that should hopefully fix your CI build) CI Passing is a requirement for merges.

**Comment by ssbarnea** [comment]
@slank I think that by now you already know what you have to fix (that one unittest). Once you pass CI we will merge it. 

I had to manually give travis a kick in order to force it to rebuild the PR after we switched from travis.org to travis.com.

**Comment by slank** [comment]
@ssbarnea @hdost sorry I didn't think to rebase. Thank you for all your help.

**Pull request merged**